### PR TITLE
feat: Import progress

### PR DESCRIPTION
**Pull request by atopper**
- show some progress (spinner, duration)
- make required fields 'required'
- do not pass 'script' if not provided
- do not query if a key was not provided

Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #<gh-issue-id>

Test URLs:
- Before: https://main--helix-labs-website--adobe.hlx.live/tools/import/index.html
- After: https://import-progress--helix-labs-website--adobe.hlx.live/tools/import/index.html

### CODE REVIEW

**Review by fkakatie — APPROVED** [review]

**Review by arumsey — APPROVED** [review]

File: tools/import/import.css [hunk 1]
```diff
@@ -123,3 +123,17 @@
...
 }
 
+.spinner {
+  display: inline-block;
+  width: 1rem;
+  height: 1rem;
+  margin: 6px 0 0 1rem;
+  border: 3px solid rgb(0 0 0 / 30%);
+  border-radius: 50%;
+  border-top-color: #ccf;
```
Comment (arumsey) [inline tools/import/import.css:133]: Is there an existing color var that could be used here?

File: tools/import/import.css [hunk 2]
```diff
@@ -123,3 +123,17 @@
   background-color: var(--gray-25);
 }
 
+.spinner {
```
Comment (arumsey) [inline tools/import/import.css:126]: Candidate for `styles.css`?

File: tools/import/importservice.js [hunk 3]
```diff
@@ -62,7 +62,7 @@ export default class ImportService {
     const $this = this;
 
     const poll = async () => {
-      if ($this.busy) {
+      if ($this.busy || !$this.apiKey) {
```
Comment (arumsey) [inline tools/import/importservice.js:65]: I think this check should be moved further down.

Comment (arumsey) [reply tools/import/importservice.js:65]:
```
if (this.config.poll && this.apiKey)
```

Comment (atopper) [reply tools/import/importservice.js:65]: Has to be before line 70.  Or do you mean inside "queryJob"?

File: tools/import/import.css (hunk 2 above)
Comment (atopper) [reply tools/import/import.css:126]: I'm not sure.  Wouldn't I have to make it more generic for everyone (flexible height, width, border width, margins, colors variants, etc.).

File: tools/import/importservice.js (hunk 3 above)
Comment (arumsey) [reply tools/import/importservice.js:65]: `poll` is an inner function. Add the check on Line 88 before `await poll();` is called for the first time.

**Review by arumsey — APPROVED** [review]

**Pull request merged**